docs/01-app/03-api-reference/01-directives/use-cache-remote.mdx

```diff
@@ -0,0 +1,557 @@
+| ------------------------------- | ----------------------------------- | --------------------------------------------------------- | ----------------------------------- |
+| **Works in dynamic context**    | No (requires static context)        | Yes (designed for dynamic contexts)                       | Yes                                 |
+| **Access to `await cookies()`** | No                                  | Yes (but not recommended - use `private` instead)         | Yes                                 |
+| **Access to `await headers()`** | No                                  | Yes (but not recommended - use `private` instead)         | Yes                                 |
+| **After `await connection()`**  | No (won't cache)                    | Yes (will cache)                                          | No                                  |
+| **Stored in cache handler**     | Yes (server-side)                   | Yes (server-side)                                         | No (client-side only)               |
+| **Cache scope**                 | Global (shared)                     | Global (shared)                                           | Per-user (isolated)                 |
+| **Runtime pre-fetchable**       | N/A (pre-rendered at build)         | No                                                        | Yes (when configured)               |
+| **Use case**                    | Static, shared content (build-time) | Dynamic, shared content in runtime contexts (per-request) | Personalized, user-specific content |
+
+### When to use each
+
+Choose the right caching directive based on your use case:
+
+**Use [`use cache`](/docs/app/api-reference/directives/use-cache) when:**
+
+- Content can be prerendered at build time
+- Content is shared across all users
+- Content doesn't depend on request-specific data
+
+**Use `'use cache: remote'` when:**
+
+- You need caching within dynamic context
+- Content is shared across users but must be rendered per-request
+- You want to cache expensive operations in a server-side cache handler
+
+**Use [`'use cache: private'`](/docs/app/api-reference/directives/use-cache-private) when:**
+
+- Content is personalized per-user (depends on cookies, headers)
+- You need [runtime prefetching](/docs/app/guides/prefetching) of user-specific content
+- Content should never be shared between users
+
+## How it works
+
+The `'use cache: remote'` directive enables runtime caching of shared data in dynamic contexts by storing results in server-side cache handlers rather than prerendering at build time.
+
+### Dynamic context detection
+
+When Next.js encounters certain APIs like [`connection()`](/docs/app/api-reference/functions/connection), [`cookies()`](/docs/app/api-reference/functions/cookies), or [`headers()`](/docs/app/api-reference/functions/headers), the context becomes "dynamic". In a dynamic context:
+
+1. **Regular `use cache` stops working** - it won't cache anything
```

Comment:
Maybe mention that it'll error in development? Because the fear is that this will happen silently.